runtime/error.c

```diff
@@ -1,9 +1,138 @@
+#include <string.h>
+#include <stdlib.h>
+#include <stdio.h>
+
 #include "balrt.h"
+#include "third-party/libbacktrace/backtrace.h"
+
+#define INITIAL_FRAME_COUNT 1
+#define MAX_FRAME_COUNT 200
+#define MAX_BYTES_COUNT INT32_MAX
+#define INITIAL_BYTES_COUNT 8
+#define SKIP_FROM_END 1 // This number counts from the last stack frame
+#define THREAD 0
+
+typedef struct {
+    uint32_t nFrames;
+    uint32_t cFrames;
+    uint32_t nBytes;
+    uint32_t cBytes;
+    Frame *frames;
+    char *bytes;
```

Comment:
I would have written this to have a single monolithic block of memory.
```C
typedef struct {
    uint32_t nFrames;
    void *bytes;
} Trace;
```

```
+--+------+---------------+----------------+--+------+----------...
|pc|lineNo|func name .../0| file name .../0|pc|lineNo|.............
+--+------+---------------+----------------+--+------+----------...
```
But I am not sure if this is better than what you have done here. You have the advantage of being able access things as struct fields (eg `frame->lineno`) which is nice, but having two block, one referring to other is not looking nice to me.

@jclark any preference ?

Reply:
I think what he has done is fine (I suggested it). The blocks are saved together in one allocation in the Error object. Doing it like this makes for significantly simpler code: no need to manually pad out the strings to an appropriate alignment boundary.

Reply:
Other big advantage of this approach is that filenames/function names can be shared between frames.  Often many frames in a backtrace will have the same function name and/or filename.

Reply:
The code should scan a few frames to try to see if it can reuse a filename/function name.